### PR TITLE
refactor: Update echobot

### PR DESCRIPTION
**Pull request by nurupo**
- Added working bootstrap nodes
- Removed the useless 4th member of the `DHT_node` struct
- Updated to [the full example from wiki](https://wiki.tox.chat/developers/client_examples/echo_bot) instead of the minimal one (more functions + creates toxsave file)
- Astyled with c-toxcore's rules
- No warnings with `-Wall -Wextra -pedantic -Wno-unused-parameter`

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/toxins/20)

### CODE REVIEW

File: echobot/echobot.c [hunk 1]
```diff
@@ -18,22 +18,111 @@ typedef struct DHT_node {
     const char *ip;
     uint16_t port;
     const char key_hex[TOX_PUBLIC_KEY_SIZE * 2 + 1];
-    unsigned char key_bin[TOX_PUBLIC_KEY_SIZE];
 } DHT_node;
 
+const char *savedata_filename = "savedata.tox";
```
Comment (iphydf) [inline echobot/echobot.c:23]: `static const char *const savedata_filename`

File: echobot/echobot.c [hunk 2]
```diff
@@ -18,22 +18,111 @@ typedef struct DHT_node {
...
 void friend_request_cb(Tox *tox, const uint8_t *public_key, const uint8_t *message, size_t length,
                        void *user_data)
 {
     tox_friend_add_norequest(tox, public_key, NULL);
+
+    update_savedata_file(tox);
 }
 
-void friend_message_cb(Tox *tox, uint32_t friend_number, Tox_Message_Type type, const uint8_t *message,
+void friend_message_cb(Tox *tox, uint32_t friend_number, TOX_MESSAGE_TYPE type, const uint8_t *message,
```
Comment (iphydf) [inline echobot/echobot.c:119]: Was this change intentional?

Comment (nurupo) [reply echobot/echobot.c:119]: No, will fix that. I have updated https://wiki.tox.chat/developers/client_examples/echo_bot which I have back-ported here, but I guess I also hanged back those types my mistake when copy-pasting.

File: echobot/echobot.c (hunk 1 above)
Comment (nurupo) [reply echobot/echobot.c:23]: What about functions being non-static or `Tox *create_tox()` not being `Tox *create_tox(void)`,
or not checking return values of almost all functions that can fail, or using  [non portable](https://wiki.sei.cmu.edu/confluence/display/c/FIO19-C.+Do+not+use+fseek%28%29+and+ftell%28%29+to+compute+the+size+of+a+regular+file) `fseek(f, 0, SEEK_END);`, assigning an int to a char `tox_id_hex[i] = toupper(tox_id_hex[i]);` , or...

well, there are many things wrong with this code, however that's not the point. The point is for it to be readable, with as little unnecessary error checking, typing and casting clutter as possible, to serve as a tutorial on how to get started with tox and write a simple tox client.

Comment (nurupo) [reply echobot/echobot.c:23]: Well, at least that's the point of it on the wiki article, and the echobot was taken here from the wiki, so i'd like to keep them in sync.
